### ISSUE TITLE
Prospector changes the behaviour of pylint for what concerns absolute imports

### ISSUE THREAD

**Issue by giovannipizzi** [PyCQA/prospector#294]
In particular, if I am using a globally-installed module (e.g.: `tzlocal`) but I also create a folder called `tzlocal` in my source tree, prospector will complain (because it will try to load the `tzlocal` module in the path rather than the global one when it encounters `import tzlocal` in a python file in the same folder).
A repository with a minimal example to reproduce the problem (thanks to @ltalirz) is here:
https://github.com/ltalirz/prospector-pylint-absolute-import

And we are having issues e.g. in our repo that uses prospector in pre-commit hooks, see e.g. aiidateam/aiida_core#2205

In particular:
- if I do `pylint path/to/file` absolute imports are properly honored
- Instead if I do `prospector path/to/file` all imports are always considered as relative

Apparently, this is done on purpose?
https://github.com/PyCQA/prospector/blob/744112f8a837bf48af1f88e66a33e7dfe2368d85/prospector/tools/pylint/__init__.py#L147-L156

However, I think this is a problem because it changes the behaviour of pylint.

A (partial) workaround would be to just call `prospector` or `prospector .` (note instead `prospector pkg` would give the unexpected/unwanted behaviour described above).

However, this is suboptimal and not a good workaround: it makes it not possible to use it with pre-commit hooks, where we can run prospector only on those few files that were changed.

One possible suggestion I have is to add a command line flag to specify the path to use (or equivalently not to extend the path with the folders of each file specified on the command line).

Another option is *not* to deprecate the `-p` option on the command line, but still honour the remaining arguments (now called checkpaths) to whitelist files to be checked.

I.e., what I mean is:

`prospector -p . pkg/file1.py`

should not add `pkg` to the sys.path, and check *only* the file `pkg./file1.py`.

Note that now, if `-p` is specified, all the paths specified (`pkg/file1.py` in this example) are ignored, and therefore *all* files in the package are checked.

It would be great to have this feature, because right now we would need either to disable pylint errors one each file, or to refactor our code to change all folder names, that looks suboptimal...

**Comment by giovannipizzi** [comment]
@carlio @chocoelho it would be great to have your feedback on this to know how to proceed. Thanks!

**Comment by donbowman** [comment]
i find I cannot use prospector with poetry with pylint because of this.
pylint always looks in the system paths for the imports, doesn't find the ones in the venv.

**Comment by chocoelho** [comment]
> i find I cannot use prospector with poetry with pylint because of this.
> pylint always looks in the system paths for the imports, doesn't find the ones in the venv.

Hi all, I'll check this out to be part of our next release. Should be able to dive through this over the weekend.

**Comment by chocoelho** [comment]
@donbowman @giovannipizzi Hey there, in order to keep the current behavior as @carlio had a reasoning for using this approach back then, I'll use an approach that allows such behavior to exist and still be the default, while using some `pylint_default_path_finder` be set in the configuration so that the default pylint approach be allowed to be used.

**Comment by chocoelho** [comment]
@donbowman @giovannipizzi just release 1.3.0.dev6 containing a fix for this.
In order to preserve current prospector behavior and allow the usage of how pylint would handle sys.paths by default, I added support for a new option `use_pylint_default_path_finder` that can be used in the following way:

```yaml
pylint:
  options:
    use_pylint_default_path_finder: True
```

**Comment by chocoelho** [comment]
Please, try this new version as this should be the last fix before 1.3.0 release.

**Comment by donbowman** [comment]
ok will try tomorrow.

**Comment by donbowman** [comment]
ok i can confirm based on a trivial test that it does what i expected.

In pyproject.toml:
```
[tool.poetry.dev-dependencies]
prospector = { git = "https://github.com/PyCQA/prospector", tag = "1.3.0.dev6" }
```

poetry run prospector somefile

it uses the flask et al packages in the venv.

**Comment by carlio** [comment]
Honestly I don't remember the original reasoning, it was like 5 years and many pylint and python versions ago! Thanks for adding this @chocoelho :-)

**Comment by chocoelho** [comment]
Great! I'll review https://github.com/PyCQA/prospector/pull/385 once more later today to see if there's anything else to fix and hopefully, tomorrow 1.3.0 should be released :tada:

**Comment by giovannipizzi** [comment]
Thanks!
